vscode-integration/src/glsp-commands.ts

```diff
@@ -32,6 +32,9 @@ export namespace GLSPCommand {
     export const CENTER = 'center';
     export const DELETE = 'delete';
     export const LAYOUT = 'layout';
+    export const NEXT_NODE = 'nextNode';
```

Comment:
The navigation commands are diagram-specific and should be defined in the workflow-example package.

Reply:
I did my best trying to extract the registration of commands and their actions into user space. Feel free to feedback whether this is the direction we want to move in.

Reply:
> I am a bit stuck regarding the external navigation. For some reason the `NavigateToExternalTargetAction` never gets handed through to a place where we can interact with the VSCode api.
> 
> The only way I found to intercept the Action is to bind an action handler in the `GLSPStarter` via the `configureActionHandler` method (See [dacd7af](https://github.com/eclipse-glsp/glsp-vscode-integration/commit/dacd7afed4257911ad6c68c476ecfa2c4387e8b6)). The problem now I believe is, that we are in the webview space and can't access the vscode api from inside. When I try to import `vscode` in the action handler I get a transpilation error that the module can't be found (even though I installed). I believe I am missing something obvious here...
> I would be very greatful if you had any ideas on how to solve this.

You are right, this looks like a major limitation in our action handling flow. There is currently no possibility to handle a local webview action (i.e. a action that was dispatched from within glsp/sprotty and is not delegated to the glsp server) in the vscode extension. However, the vsode extension is the only place where you have access to the vscode API. As you already noticed you cannot use the vscode API inside the webview, even if you import it. I have created https://github.com/eclipse-glsp/glsp/issues/234 to track this and I'm already working on a fix.

Reply:
Thanks for looking into this! I'm glad my attempts to get this working were not completely in vain.

Reply:
I have open a [PR](https://github.com/eclipse-glsp/glsp-vscode-integration/pull/6) to fix this #6. I tested this with a mock `ExtensionActionHandler` for the `NavigateToExternalTargetAction` and everything seems to work as expected now.